src/NServiceBus.RavenDB/Internal/ConnectionVerifier.cs

```diff
@@ -1,112 +1,111 @@
 ﻿namespace NServiceBus.RavenDB.Internal
 {
     using System;
+    using System.Diagnostics;
     using System.IO;
     using System.Net;
-    using System.Text;
     using NServiceBus.Logging;
     using Raven.Client;
     using Raven.Imports.Newtonsoft.Json;
 
     class ConnectionVerifier
     {
-        const string WrongRavenVersionMessage =
-            @"The RavenDB server you have specified is detected to be {0}. NServiceBus requires RavenDB version 3.0 build 3660 (or a higher build number for version 3.0) to operate correctly. Please update your RavenDB server.
-Further instructions can be found at: http://particular.net/articles/using-ravendb-in-nservicebus-installing";
 
-        static readonly ILog Logger = LogManager.GetLogger(typeof(ConnectionVerifier));
+        static ILog Logger = LogManager.GetLogger(typeof(ConnectionVerifier));
 
         internal static void VerifyConnectionToRavenDBServer(IDocumentStore store)
         {
-            RavenBuildInfo ravenBuildInfo = null;
-            var connectionSuccessful = false;
-            Exception exception = null;
-            try
+            Version serverVersion;
+            if (!TryGetServerVersion(store, out serverVersion))
+            {
+                return;
+            }
+            var clientVersion = GetClientVersion();
+            if (AreVersionsCompatible(serverVersion, clientVersion))
             {
-                store.Initialize();
+                Logger.InfoFormat("Connection to RavenDB at {0} verified. Server version: {1}", store.Url, serverVersion);
+                return;
+            }
+            var message = string.Format(
```

Comment:
I would also include what the user needs to do to fix this, eg
"Please upgrade your server version to x or higher",
and maybe also point them to doco about server upgrade ?